### PR TITLE
Discount superhero

### PR DESCRIPTION
**Pull request by Arivias**
[FEATURE]

## About the PR 
Adds the discount superhero spooktober latejoin job.
DS starts with:
- A fancy costume
- security access (but no equipment to try to avoid encroaching _too_ much on the regular security officer job)

DS gets a random set of mutations from the following lists:
one non-lethal incapacitator:
- eyebeams (new stun variant)
- empowered cryokinesis
- weaponized vestigial ballistics (target knocked down for a few seconds)

one resistance (replaced with regenerator if vestigial ballistics is picked):
- fire
- cold
- rad
- oxy

plus adrenaline rush, as suggested by kyle.

DS cannot be a traitor/spy/etc.

## Why's this needed? 
Suggested by Mouse on the forum.

Provides a fun Spooktober themed latejoin security job.

## Changelog

```
(u)Sidewinder7:
(*)Added the Discount Superhero latejoin job.
```

### CODE REVIEW

**Comment by Kyle2143** [comment]
This is good, but I don't see any HALLOWEEN defines that specify that this role is only for Spooktober.  Also, it could probably have another useful gene, to make it superhero-y like adrenaline rush. Kinda OP, but hey, it's a superhero.

**Comment by Arivias** [comment]
I've had a few more ideas since I posted this. I'll see what I can do.

File: code/datums/jobs.dm [hunk 1]
```diff
@@ -2007,7 +2007,32 @@
...
 			return
-		M.bioHolder.AddEffect("eyebeams")
+		M.traitHolder.addTrait("training_security")
+		if(prob(60))
+			var/aggressive = pick("eyebeams","cryokinesis")
+			var/defensive = pick("fire_resist","cold_resist","food_rad_resist","breathless") // no thermal resist, gotta have some sort of comic book weakness
+			var/datum/bioEffect/power/be = M.bioHolder.AddEffect(aggressive, magical=1)
+			if(aggressive == "eyebeams")
+				var/datum/bioEffect/power/eyebeams/eb = be
+				eb.stun_mode = 1
```
Comment (Sovexe) [inline code/datums/jobs.dm:2017]: this should probably still be marked as `altered` to prevent splicing

File: code/datums/jobs.dm [hunk 2]
```diff
@@ -2007,7 +2007,32 @@
...
+			else
+				be.power = 1
+				be.altered = 1
+				be.safety = 1
```
Comment (Sovexe) [inline code/datums/jobs.dm:2021]: empowered, synchronized, stabilized genes that can't be cured by mutadone or scrambled away by genetics exceeds what I would consider "discount" that's more of an actual super hero

File: code/datums/jobs.dm [hunk 3]
```diff
@@ -2007,7 +2007,32 @@
...
+			sl.safety = 1
+			sl.altered = 1
+			sl.cooldown = 300
+			sl.stun_mode = 1
+			var/datum/bioEffect/regenerator/r = M.bioHolder.AddEffect("regenerator", magical=1)
+			r.regrow_prob = 10
+		var/datum/bioEffect/power/be = M.bioHolder.AddEffect("adrenaline", magical=1)
+		be.safety = 1
+		be.power = 1
+		be.altered = 1
```
Comment (Sovexe) [inline code/datums/jobs.dm:2034]: likewise

**Review by Sovexe — COMMENTED** [review]
I would also consider the possibility of tying the genetic effects to the suit itself. Such that they are lost when you remove the suit and regained when equipping it. Don't have to do this, just an idea.

**Review by Sovexe — APPROVED** [review]

File: code/modules/medical/genetics/bioEffects/powers.dm [hunk 4]
```diff
@@ -2145,7 +2160,8 @@
 					//double power if the ability is empowered (doesn't really do anything, but w/e)
 					var/tmp_force = thrown_limb.throwforce
 					thrown_limb.throwforce = limb_force* (throw_power+1)	//double damage if empowered
-					thrown_limb.throw_at(target, range, throw_power * (linked_power.power+1))
+					var/callback = (SL && SL.stun_mode) ? /datum/targetable/geneticsAbility/shoot_limb/proc/hit_callback : null
```
Comment (Sovexe) [inline code/modules/medical/genetics/bioEffects/powers.dm:2163]:
```suggestion
					var/callback = (SL?.stun_mode) ? /datum/targetable/geneticsAbility/shoot_limb/proc/hit_callback : null
```

**Review by Sovexe — CHANGES_REQUESTED** [review]

**Review by Sovexe — APPROVED** [review]

**Pull request merged**